src/gcp/cloudscheduler.ts

```diff
@@ -12,38 +14,116 @@ export interface Schedule {
+/**
+ * Gets a cloudScheduler job with the given name.
+ * If no job with that name exists, this will return a 404.
+ * @param name The name of the job to get.
+ */
+export function getJob(name: string): Promise<any> {
+  return api.request("GET", `/${VERSION}/${name}`, {
+    auth: true,
+    origin: api.cloudschedulerOrigin,
+  });
+}
+
+/**
+ * Updates a cloudScheduler job.
+ * Returns a 404 if no job with that name exists.
+ * @param job A job to update. Note that name cannot be updated.
+ */
+export function updateJob(job: Job): Promise<any> {
+  return api.request("PATCH", `/${VERSION}/${job.name}`, {
+    auth: true,
+    origin: api.cloudschedulerOrigin,
+    data: Object.assign({ timeZone: DEFAULT_TIME_ZONE }, job),
+  });
+}
+
+/**
+ * Checks for a existing job with the given name.
+ * If none is found, it creates a new job.
+ * If one is found, and it is identical to the job parameter, it does nothing.
+ * Otherwise, if one is found and it is different from the job param, it updates the job.
+ * @param job A job to check for and create, replace, or leave as appropriate.
+ * @throws { FirebaseError } if an error respnse other than 404 is recieved on the GET call.
+ */
+export async function createOrReplaceJob(job: Job): Promise<any> {
+  const jobName = `${job.name.split("/")[5]}`;
   try {
-    await createJob(schedule);
-    logLabeledSuccess("functions", `created scheduler job ${jobName}`);
+    const existingJob = await getJob(job.name);
+    if (!job.timeZone) {
+      // We set this here to avoid recreating schedules that use the default timeZone
+      job.timeZone = DEFAULT_TIME_ZONE;
+    }
+
+    if (isIdentical(existingJob.body, job)) {
+      logLabeledBullet("functions", `scheduler job ${jobName} is up to date, no changes required`);
+      return;
+    } else {
+      const updatedJob = await updateJob(job);
+      logLabeledBullet("functions", `updated scheduler job ${jobName}`);
+      return updatedJob;
+    }
   } catch (e) {
-    if (e.context.response.statusCode !== 409) {
+    // If the error status is 404, no job exists, so we can create one
+    // If it is anything else, we should error out
+    if (e && e.context && e.context.response && e.context.response.statusCode !== 404) {
       throw e;
     }
-    logLabeledBullet("functions", `re-creating scheduler job ${jobName}`);
-    await deleteJob(schedule.name);
-    return createJob(schedule);
+    const newJob = await createJob(job);
+    logLabeledSuccess("functions", `created scheduler job ${jobName}`);
+    return newJob;
   }
 }
+
+/**
+ * Check if two jobs are functionally equivalent.
+ * @param existingJob a job to compare.
+ * @param newJob a job to compare.
+ */
+function isIdentical(existingJob: Job, newJob: Job): boolean {
+  return (
+    existingJob &&
+    existingJob.schedule === newJob.schedule &&
+    existingJob.timeZone === newJob.timeZone &&
+    isEqual(existingJob.retryConfig, newJob.retryConfig)
```

Comment:
I'm pretty used to seeing lodash imported as `_`, which makes these methods more recognizable. I'm a bit torn on whether to keep using `_` which is less ideal, but standard, or to switch to importing the method you need as you did here. I think I'm leaning toward using `_` since it creates a strong signal that this function is on lodash and is not coming from elsewhere in this file, which is what I originally thought.

Reply:
Yeah, I was torn on this as well - I think I went with this because I had just read a doc about minimizing imports. I'm gonna switch this back since consistency/clarity is more important than a minor efficiency improvement. We might want to switch to an individual import style across the project in another refactor in the future tho.

Reply:
👍